### ISSUE TITLE
Measurement results depend on node-ids

### ISSUE THREAD

**Issue by easbar** [graphhopper/graphhopper#2121]
Currently we draw random source/target nodes for our sample queries in the `Measurement` class. However, this means our results depend on the node IDs. For example without `optimize()` we get different queries/results: #2118

Instead, we should:

1) Generate random GPS coordinates (within the boundaries of the measurement map) and check if we get a valid snapped node, if yes add this node to an array of query nodes. Do this until we got a certain amount of query nodes.

2) Draw random nodes from our array of query nodes to perform the routing queries.

**Comment by easbar** [comment]
Almost forgot, but another problem is that we use the random node ids to get the gps coordinates. But this means that all query graph snaps are always tower nodes, while in reality most snaps will introduce virtual nodes. Therefore its also more realistic to use random gps points.

**Comment by karussell** [comment]
To find out if the coordinates are within the map boundaries we would need to do a locationIndex lookup. Do we care about this happening twice? Or should we accept problems and only ignore them in the stats? (and if there is an unusual high error count abort the measurement?)

**Comment by easbar** [comment]
Yes we need the location index. But we can just build a list of valid points upfront. To use block area we can then just take points from this list and discard those that are within the blocked area.

**Comment by easbar** [comment]
Let's see how this affects measurement, but in any case I think this is the more 'correct' way to do it: d111819426692536021b833c47fda31b4652a78a

**Comment by karussell** [comment]
Nice, thanks for taking care of it :)
